library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -1217,7 +1215,7 @@ public void onMapReady(MapboxMap mapboxMap) {
 
     @Override
     public boolean changeLocationUpdates(long updateInterval, long fastestUpdateInterval, int accuracyLevel) {
-        if (getLocationClient() != null) {
+        if (updateInterval > -1 && fastestUpdateInterval > -1 && getLocationClient() != null) {
```

Comment:
Could you also add a check for accuracyLevel.

I also see it's not used in the function, so it can alternatively be removed?